ui/js/reducers/claims.js

```diff
@@ -157,21 +161,35 @@ reducers[types.ABANDON_CLAIM_STARTED] = function(state, action) {
 };
 
 reducers[types.ABANDON_CLAIM_SUCCEEDED] = function(state, action) {
-  const { claimId } = action.data;
+  const { claimId, txid, nout } = action.data;
   const myClaims = new Set(state.myClaims);
   const byId = Object.assign({}, state.byId);
   const claimsByUri = Object.assign({}, state.claimsByUri);
+  const supports = byId[claimId].supports;
   const uris = [];
 
-  Object.keys(claimsByUri).forEach(uri => {
-    if (claimsByUri[uri] === claimId) {
-      delete claimsByUri[uri];
-    }
-  });
+  // This logic is needed when a claim has supports
+  // and it is the support that is being abandoned
+  // so we need to remove the support from the state
+  // but this is not working, even after calling resolve on the uri.
+  if (supports && supports.length > 0) {
+    const indexToDelete = supports.findIndex(support => {
+      return support.txid === txid && support.nout === nout;
+    });
 
-  delete byId[claimId];
-  myClaims.delete(claimId);
+    supports.splice(indexToDelete, 1);
```

Comment:
I think it is fine/right to update the state in this way, but it would also be an option to just trigger a re-fetch these after an abandon.

Reply:
I'm removing this logic. And taking your suggestion of re-fetching.